src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -976,14 +985,14 @@ void GoogleCalendarSyncAdaptor::updateLocalCalendarNotebookEvents(int accountId,
         m_storage->modifiedIncidences(&updatedList, KDateTime(since), googleNotebook->uid()); // TODO: since UTC?
         Q_FOREACH(const KCalCore::Incidence::Ptr incidence, allList) {
             QString gcalId = gCalEventId(incidence);
-            KCalCore::Event::Ptr eventPtr = m_calendar->event(incidence->uid());
+            KCalCore::Event::Ptr eventPtr = m_calendar->event(incidence->uid(), incidence->hasRecurrenceId() ? incidence->recurrenceId() : KDateTime());
```

Comment:
Doesn't matter much, but I think it should be safe enough to just use recurrenceId(). Should return invalid KDateTime when not existing.